context/app/static/js/components/detailPage/provenance/utils.ts

```diff
@@ -8,6 +8,27 @@ export function createProvDataURL(uuid: string, entityEndpoint: string) {
   return `${entityEndpoint}/entities/${uuid}/provenance`;
 }
 
+function findDifference(obj1: object, obj2: object) {
+  const diffKeys: string[] = [];
+  Object.keys(obj1).forEach((key) => {
+    // @ts-expect-error -- We're specifically checking for the key in obj2
+    if (!(key in obj2) || obj1[key] !== obj2[key]) {
+      diffKeys.push(key);
+    }
+  });
+  Object.keys(obj2).forEach((key) => {
+    // @ts-expect-error -- We're specifically checking for the key in obj1
+    if (!(key in obj1) || obj1[key] !== obj2[key]) {
+      if (!diffKeys.includes(key)) {
+        diffKeys.push(key);
+      }
+    }
+  });
+  return diffKeys;
+}
```

Comment:
Just brainstorming a potentially simpler method. Feel free to take or leave.

```
const diffKeys = Object.keys({...obj1, ...obj2}).reduce((acc, k) => {
 const v1 = obj1?.[key];
 const v2 = obj2?.[key];

if(v1 === v2){
   return acc;
}

return [...acc, key];
}, [])
```

Reply:
Thanks, good call. I was definitely overthinking this.